packages/mongodb-access-adapter/datagrip-access-adapter/src/main/kotlin/com/mongodb/jbplugin/accessadapter/datagrip/adapter/DataGripMongoDbDriver.kt

```diff
@@ -81,7 +82,14 @@ internal class DataGripMongoDbDriver(
 
     override suspend fun connectionString(): ConnectionString = ConnectionString(dataSource.url!!)
 
-    override suspend fun <S> explain(query: Node<S>): ExplainPlan = ExplainPlan.CollectionScan
+    override suspend fun <S> explain(query: Node<S>): ExplainPlan {
+        val explainQuery = MongoshDialect.formatter.indexCommandForQuery(query)
+        if (explainQuery != "") {
+            return ExplainPlan.CollectionScan
+        }
+
+        return ExplainPlan.IndexScan
```

Comment:
Is this actually supposed to be used for demo or just some placeholder code?

Reply:
Only for the demo :D